### PR TITLE
feat(coverage): Demangling of Java function names in coverage results

### PR DESCRIPTION
**Pull request by vogelsgesang**
This commit adds demangling support for Java. I couldn't find any good
third-party library for this, so I quickly implemented my own, based on
https://docs.oracle.com/javase/specs/jvms/se8/html/jvms-4.html#jvms-4.3.2

It might still be have a couple of bugs, but at least for my simple
smoke tests, demangling seems to work just fine.

Works towards #362

### CODE REVIEW

**Comment by vogelsgesang** [comment]
Gentle ping 🙂

File: test/lcov_parser.test.ts [hunk 1]
```diff
@@ -96,13 +96,12 @@ describe("The lcov parser", () => {
...
       assert(initFunc !== undefined);
-      assert.equal(initFunc.name, "com/example/myproject/Greeter::<init> ()V");
+      assert.equal(initFunc.name, "void Greeter::<init>()");
       assert.equal(initFunc.executed, 1);
       const convertFunc = getFunctionByLine(fileCov, 28);
       assert.equal(
         convertFunc.name,
-        "com/example/myproject/Greeter::convertStreamToString " +
-          "(Ljava/io/InputStream;)Ljava/lang/String;",
+        "String Greeter::convertStreamToString(InputStream)",
```
Comment (cameron-martin) [inline test/lcov_parser.test.ts:104]: Do we definitely want to drop the namespace here, and only keep the type name?

Comment (vogelsgesang) [reply test/lcov_parser.test.ts:104]: I think it makes sense to drop the prefix, yes.

In the VSCode UI, the functions are grouped by file paths. See https://code.visualstudio.com/updates/v1_88#_test-coverage for a screen shot. Note how in the lower left part, you have a file system tree. If you expand one of those file names, you can see all functions inside that file.

As such, function names only need to be unique in the context of a single file. Repeating the package name in each and every function name would not add any additional information, but would clutter the UI

**Review by cameron-martin — APPROVED** [review]

File: test/lcov_parser.test.ts (hunk 1 above)
Comment (cameron-martin) [reply test/lcov_parser.test.ts:104]: Makes sense, thanks!

**Pull request merged**